ping.go

```diff
@@ -476,16 +477,36 @@ func (p *Pinger) runLoop(
 	}
 
 	timeout := time.NewTicker(p.Timeout)
+	packetTimeout := time.NewTicker(100 * time.Microsecond)
 	interval := time.NewTicker(p.Interval)
+
 	defer func() {
 		interval.Stop()
 		timeout.Stop()
+		packetTimeout.Stop()
 	}()
 
 	if err := p.sendICMP(conn); err != nil {
 		return err
 	}
 
+	go func() {
+		for {
+			select {
+			case <-packetTimeout.C:
```

Comment:
The select has only 1 case, so does it make sense to have for range over packetTimeout.C?